File: src/utils/ai_completion.py
from anthropic import Anthropic
from openai import OpenAI
from typing import Optional

class AICompletion:
    def __init__(self, client, model):
        self.client = client
        self.model = model

    def get_completion(
        self, 
        system_prompt: str, 
        user_prompt: str, 
        max_tokens: int = 2000, 
        temperature: float = 0.7
    ) -> Optional[str]:
        """Get completion from the language model with unified interface for all providers."""
        try:
            if isinstance(self.client, Anthropic):
                response = self.client.messages.create(
                    model=self.model,
                    max_tokens=max_tokens,
                    temperature=temperature,
                    system=system_prompt,
                    messages=[{
                        "role": "user",
                        "content": user_prompt
                    }]
                )
                return response.content[0].text

            elif isinstance(self.client, OpenAI):
                messages = [
                    {"role": "system", "content": system_prompt},
                    {"role": "user", "content": user_prompt}
                ]
                response = self.client.chat.completions.create(
                    model=self.model,
                    messages=messages,
                    max_tokens=max_tokens,
                    temperature=temperature
                )
                return response.choices[0].message.content

            else:
                raise ValueError(f"Unsupported client type: {type(self.client)}")

        except Exception as e:
            print(f"Error in API call: {str(e)}")
            print(f"Model: {self.model}")
            print(f"System prompt: {system_prompt[:100]}...")
            print(f"User prompt: {user_prompt[:100]}...")
            
            if hasattr(e, 'response'):
                print(f"Response status: {e.response.status_code}")
                print(f"Response body: {e.response.text}")
            
            raise 

File: nodeSrc/utils/ai_completion.js
const OpenAI = require('openai');
const { Logger } = require('./logger');
const { Config } = require('./config');

class AICompletion {
    constructor(client, model) {
        this.logger = new Logger('ai');
        const aiConfig = Config.getAIConfig();
        
        if (client) {
            this.client = client;
            this.model = model;
        } else {
            this.model = aiConfig.model || 'gpt-4';
            this.client = new OpenAI({
                apiKey: aiConfig.apiKey,
                baseURL: aiConfig.baseUrl
            });
        }
        
        console.log('AI Client initialized:', {
            model: this.model,
            hasClient: !!this.client
        });
    }

    async getCompletion(systemPrompt, userPrompt, options = {}) {
        try {
          

            const response = await this.client.chat.completions.create({
                model: this.model,
                messages: [
                    {
                        role: "system",
                        content:
                            "You are Grok, a chatbot inspired by the Hitchhiker's Guide to the Galaxy.",
                    },
                    {
                        role: "user",
                        content:
                            "What is the meaning of life, the universe, and everything?",
                    },
                ],
            });

            console.log('AI Response:', response);

            return response.choices[0].message.content;
        } catch (error) {
            this.logger.error('Error in AI completion', error);
            console.error('Detailed error:', {
                name: error.name,
                message: error.message,
                status: error.status,
                response: error.response
            });
            throw error;
        }
    }
}

module.exports = { AICompletion }; 